projects/ngneat/hotkeys/src/lib/hotkeys.service.ts

```diff
@@ -0,0 +1,73 @@
+import { DOCUMENT } from '@angular/common';
```

Comment:
Addons:

1. Add a way to add a `callback` that will be called for each `handler`:

```ts
  onShortcut(cb: (event: Event, keys: string, target: Element) => any) {
    this.callbacks.push(cb);
    return () => this.callbacks.filter(current => current !== cb);
  }

  addShortcut() {
    // inside the handler
    this.callbacks.forEach(cb => cb(event, normalizedKeys, target));
  }
```

2. Add a method that helps with the modal:

```ts

registerHelpModal(openHelpModalFn) {
 this.addShortcut({ keys: helpShortcut || 'shift.?', preventDefault: false }).subscribe(e => {
      const skipMenu =
        /^(input|textarea)$/i.test(document.activeElement.nodeName) || (e.target as any).isContentEditable;

      if (!skipMenu) {
        size(this.hotkeys) && openHelpModalFn();
      }
    });
}
```

Reply:
What are you trying to accomplish with 

```

return () => this.callbacks.filter(current => current !== cb);

```

Reply:
We need to call it in the dispose section, so we'll not have a leak